slycot/synthesis.py

```diff
@@ -1700,6 +1699,119 @@ def sb10jd(n,m,np,A,B,C,D,E,ldwork=None):
+        = 0: continuous-time system;
+        = 1: discrete-time system.
+    flag : int
+        If flag = 0, then the system zeros and poles are not constrained.
+        If flag = 1, then the system zeros and poles will have negative
+        real parts in the continuous-time case, or moduli less than 1 in
+        the discrete-time case. Consequently, flag must be equal to 1 in
+        mu-synthesis routines.
+    lendat : int
+        The length of the vectors rfrdat, ifrdat and omega.
+        length >= 2.
+    rfrdat : dimension (lendat), array_like
+        The real part of the frequency data to be fitted.
+    ifrdat : dimension (lendat), array_like
+        The imaginary part of the frequency data to be fitted.
+    omega : dimension (lendat), array_like
+        The frequencies corresponding to rfrdat and ifrdat.
+        These value must be nonnegative and monotonically increasing.
+        Additionally, for discrete-time systems they must be between 0 and PI.
+    n : int
+        On entry, the desired order of the system to be fitted.
+        n <= lendat-1.
+    tol : int, optional
+        The length of the cache array.
+    ldwork : int
+        With None it will be automatically calculated.
+        For details see SLICOT help.
+
+    Returns
+    -------
+    n : int
+        The order of the obtained system. The value of n
+        could only be modified if n > 0 and flag = 1.
+    A : (n, n) ndarray
+        The computed matrix A.
+        matrix A.
+    B : (n, 1) ndarray
+        The computed column vector B.
+    C : (1, n) ndarray
+        The computed row vector C.
+    D : (1, 1) ndarray
+        The computed scalar D.
+
+    Raises
+    ------
+    SlycotArithmeticError
+        :info == 0:  successful exit;
+        :info < 0:   if info = -i, the i-th argument had an illegal value
+        :info = 1:   if the discret --> continous transformation cannot be made;
+        :info = 2:   if the system poles cannot be found;
+        :info = 3:   if the inverse system cannot be found, i.e., D is (close to) zero;
+        :info = 4:   if the system zeros cannot be found;
+        :info = 5:   if the state-space representation of the new transfer function T(s) cannot found;
+        :info = 6:   if the continous --> discrete transformation cannot be made.
+            The iteration for computing singular value
+            decomposition did not converge.
+    """
+
+    hidden = ' (hidden by the wrapper)'
+    arg_list = ['discfl', 'flag', 'lendat', 'rfrdat', 'ifrdat', 'omega', 
+                'n', 'A', 'lda' + hidden, 'B', 'C', 'D', 
+                'TOL', 'IWORK' + hidden, 'DWORK' + hidden, 'LDWORK',
+                'ZWORK' + hidden, 'LZWORK', 'INFO' + hidden]
+
+    if ldwork is None:
+        lw1 = 2*lendat + 4*2048
+        lw2 =   lendat + 6*2048
+        mn  = min(2*lendat,2*n+1)
+        if n > 0:
+            lw3 = 2*lendat*(2*n+1) + max(2*lendat,2*n+1) + max(mn+6*n+4,2*mn+1)
+        elif n == 0:
```

Comment:
Could you please cover this case in the unit tests?

Reply:
I have to look into that.